components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/RolesResource.java

```diff
@@ -134,6 +147,28 @@ protected JSONObject handleGet(MessageContext messageContext) throws UserStoreEx
         return jsonBody;
     }
 
+    private static List<String> getSearchResults(MessageContext messageContext, String searchKey) throws UserStoreException {
+        if (!Utils.isUserAuthenticated(messageContext)) {
+            LOG.warn("Listing user roles without authenticating/authorizing the request sender. Adding "
+                    + "authentication and authorization handlers is recommended.");
+        }
+        String[] roles = Utils.getUserStore(null).getRoleNames();
```

Comment:
if the user is unauthenticated shouldn't we perform the rest of the functions?

Reply:
followed the original implementaion in handleGet()

Reply:
@dulanjalidilmi

Reply:
We should not return results if the user is not authenticated. Shall we correct this? This is a mistake in previous implementation.  
@GDLMadushanka WDYT?

Reply:
We can remove the warn log since its not reacing here if not authenticated.